tools/tesseratos/include/tesseratos/toolbox/plugin.hpp

```diff
@@ -0,0 +1,46 @@
+/// @dir
+/// @brief @ref toolbox-plugin plugin directory.
+
+/// @file
+/// @brief Plugin entry point.
+/// @ingroup toolbox-plugin
+
+#pragma once
+
+#include <cubos/engine/cubos.hpp>
+
+namespace tesseratos
+{
+    /// @defgroup tesseratos-toolbox-plugin Toolbox
+    /// @ingroup tesseratos
+    /// @brief Adds a resource used to keep track of whether each tool is open or not.
+    ///
+    ///
+    /// ## Resources
+    /// - @ref Toolbox
+
+    /// @brief Resource which manages other tools windows.
+    class Toolbox final
+    {
+    public:
+        bool isOpen(std::string toolName);
+
+        void open(std::string toolName);
+
+        void close(std::string toolName);
+
+        void doClick(std::string toolName);
+
+        auto begin();
+
+        auto end();
```

Comment:
These methods are actually templated methods (auto is just a shorter way to write a template), and thus for these types specifically you have to define them here. Otherwise, if someone outside toolbox/plugin.cpp tries to use this method, there will be an error, as the return type is unknown and the function isn't defined.

```suggestion
        auto begin()
        {
            return mToolsMap.begin();
        }

        auto end()
        {
            return mToolsMap.end();
        }
```